framework/include/executioners/FEProblemSolve.h

```diff
@@ -51,4 +59,11 @@ class FEProblemSolve : public NonlinearSolveObject
   /// member in this way allows for easy boolean operations, e.g. if (_num_grid_steps)
   /// as opposed to if (_num_grids)
   const unsigned int _num_grid_steps;
+
+  /// Whether we are using fixed point iterations for multi-system
+  const bool _using_multi_sys_fp_iterations;
+  /// Convergence object to assess the convergence of the multi-system fixed point iteration
+  Convergence * _multi_sys_fp_convergence;
```

Comment:
If we had a class dedicated to multisystem solves then this could be a reference. Trying to retro-fit `FEProblemSolve` as a multi-system solve really is feeling like a poor fit. 

On a different note, I do not know if there are segregated solve options that do not fit in a fixed-point class of algorithms? In which case there may not be a need for this `_fp_` denotion

Reply:
This also takes it down the path of objects like `IterationAdaptiveDT` where we just keep bolting on pieces instead of having a modular system. What I am asking is that we have syntax for solve objects, e.g. executors

Reply:
Maybe @joshuahansel wants to work with @loganharbour to finish executors? He knocked down convergence objects

Reply:
Feels more like they knocked _me_ down 😁  I'd be happy to try to help with executors, but Sockeye deadlines demand my attention for at least two months.

Reply:
I think in a world of executors and not this current solve object paradimg, it would be quite easy to implement multisystem. You just have a MultiSystem executor which logically would have some kind of convergence object that works naturally with multisystem. Additionally the MultiSystem or MultiXXX executor would have a parameter that takes a vector of other executors to execute. Some of these may be SingleSystem executors, for whom such a vector parameter would not make sense and for whom a multi-system or multi-executor convergence object also doesn't make sense. A MultiXXX object is fundamentally different thatn a SingleXXX object because the MultiXXX object immediately introduces the possibility of iteration. I think there should be clear distinction between objects with MultiXXX functionality and SingleXXX functionality because the parameter set is logically much more complex for the former.

Reply:
> I think in a world of executors and not this current solve object paradigm, it would be quite easy to implement multisystem. You just have a MultiSystem executor which logically would have some kind of convergence object that works naturally with multisystem

Agree

> Additionally the MultiSystem or MultiXXX executor would have a parameter that takes a vector of other executors to execute. Some of these may be SingleSystem executors, for whom such a vector parameter would not make sense and for whom a multi-system or multi-executor convergence object also doesn't make sense

Agree

> A MultiXXX object is fundamentally different thatn a SingleXXX object because the MultiXXX object immediately introduces the possibility of iteration. I think there should be clear distinction between objects with MultiXXX functionality and SingleXXX functionality because the parameter set is logically much more complex for the former.

Agree for executors. And for solveObjects I am moving FEProblemSolve from a singleXXX to a MultiXXX, and the parameters are adapted to be vectors.

Reply:
@lindsayad Seems like a good design at first glance - are you requesting that executors be finished before this PR? Or just discussing future refactoring?

Reply:
> are you requesting that executors be finished before this PR

I suppose not. I've been reminded of our slack history where I agreed to kick the executor can down the road. This has been a good discussion. I think I'm fine with what we're doing here. I think that I will volunteer to help @loganharbour finish executors this FY if he wants the help since I do have the bandwidth

Reply:
Before I resolve this conversation, can we do this?
```suggestion
  /// Convergence object to assess the convergence of the multi-system fixed point iteration
  const Convergence * _multi_sys_fp_convergence;
```

Reply:
we cant, calling "isConverged" is not const

Reply:
Just tried to remind myself why it's not const. `DefaultNonlinearConvergence` and `ReferenceResidualConvergence` are big offenders. A simple example is the ping pong iteration counter, which can get incremented in a call to `checkConvergence`.

Reply:
okie dokie